### ISSUE TITLE
Tiny compressed blob utility

### ISSUE THREAD

**Issue by pepyakin** [paritytech/substrate#10161]
Sometimes there is a need to examine a compressed blob e.g. you extract from the chain. Typically, that's a wasm file.  However, chances are regular wasm tools won't work on it. This is because we compress such a blob with our simple format and the wasm tools do not know how to work with it.

To fix that, we could provide a tiny utility that would solve the issue. The utility will enable the user to compress and decompress such files. 

Finer points:

- The algorithm is trivial, it could be implemented in bash, but we can do that in Rust. The reference: [compression](https://github.com/paritytech/substrate/blob/7e42d7788002eadd94d3ce7782986a3fc6924e19/primitives/maybe-compressed-blob/src/lib.rs#L92) and [decompression](https://github.com/paritytech/substrate/blob/7e42d7788002eadd94d3ce7782986a3fc6924e19/primitives/maybe-compressed-blob/src/lib.rs#L81).
- Ideally, this tool could be installed. `cargo install` may be a good way to do that since it seems that it should be available for the majority of the target users. Maybe it would be better to put into another repo to make the install quick.
- It's possible that the file is already decompressed, in such a case we probably want to put a warning into stderr.
- This utility should be able to dump into stdout, but only if requested.

**Comment by xlc** [comment]
I guess this can be part of subwasm @chevdor

**Comment by chevdor** [comment]
Indeed, that can be added to [`subwasm`](https://github.com/chevdor/subwasm). `subwasm` already decompress the runtime as needed internally but I did not yet expose any subcommand to let users play with that as I missed a valid use case so far.

@pepyakin if you need a solution working **now**, we may use subwasm prior to `0.15.0`. It contains a bug you will find convenient as it decompress the runtime, no matter what, before storing it locally.

So using older versions and running `subwasm get --chain westend` will fetch the compressed wasm, decompress it and store it locally, making it available for you to play.

I could probably add a flag to the `get` subcommand but IMO the `get` command should remain true to whatever is on chain. I open [an issue in the subwasm repo](https://github.com/chevdor/subwasm/issues/28) and we can discuss options there.

**Comment by chevdor** [comment]
Hi @pepyakin, I don't have the "output to stdout" but you can already find a quick and dirty running version at https://github.com/chevdor/subwasm/tree/wk-zip

Would that work for your use case ? 

Btw, why do you need the result to stdout ?
Could you provide an example ? I guess you want to pipe or redirect to another tool and I would like to make sure it works fine and avoid issues like [this one](https://github.com/rust-lang/rust/issues/46016).

**Comment by pepyakin** [comment]
Oh, this is good to hear, and yeah, subwasm is good enough.

I wrote this issue, because I bumped into this situation several times when I had to inspect some wasm file. 

In some of those cases, I wanted to unpack the file, do some transformations and then pack it again. The simplest example of that is when I needed to upgrade the runtime. I did not care about the contents nor version. I just needed a different hash. For that I could just add a simple dummy section at the end. Such a section would be ignored. But in order to do that, I needed this unpacking.

Output into stdout is by no means necessary. Sometimes it would allow to do piping, but using files is good enough.

Given that PR I think we can close the issue. Thanks @xlc and @chevdor.
